src/create-glamorous.js

```diff
@@ -42,26 +45,29 @@ function createGlamorous(splitProps) {
        */
       const GlamorousComponent = withTheme(
         (props, context) => {
-          /* eslint no-use-before-define: 0 */
+          const updateClassName = shouldUpdate(props, context)
+
           const {toForward, cssOverrides, cssProp} = splitProps(
             props,
             GlamorousComponent,
           )
 
           // create className to apply
-          const className = getGlamorClassName({
-            styles: GlamorousComponent.styles,
-            props,
-            cssOverrides,
-            cssProp,
-            context,
-            displayName: GlamorousComponent.displayName,
-          })
+          GlamorousComponent.className = updateClassName ?
```

Comment:
Actually, it's a little hacky, but we might be able to change [this line](https://github.com/paypal/glamorous/blob/6f6803f15cf2df66a7b035506f5206edff75222b/src/with-theme.js#L70) from this:

```
return ComponentToTheme({...this.props, ...this.state}, this.context)
```

To this:

```
return ComponentToTheme.call(this, {...this.props, ...this.state}, this.context)
```

Then we'd change [the glamorous render function](https://github.com/paypal/glamorous/blob/6f6803f15cf2df66a7b035506f5206edff75222b/src/create-glamorous.js#L47) from an arrow function to a regular function and could reference `this` inside the glamorous function.

A bit hacky, but with comments I think it'd be ok...